### ISSUE TITLE
Cannot talk to ZMQ, connect failed

### ISSUE THREAD

**Issue by beckerdo** [FRiCKLE/ngx_zeromq#5]
I am running Nginx with the ngx_zeromq module, and I believe everything is built and running to some degree. I can get resources from my Nginx server, but when I attempt to hit the ZMQ proxy I get an error.

My configuration is identical to the example given in the readme,md:

```
http {
    upstream blackhole {
        zeromq_remote   REQ tcp://127.0.0.1:5555;
    }

    server {
        location / {
            proxy_pass  http://blackhole;
        }
    }
}
```

The error I am getting from the error.log is:

```
2014/05/28 11:06:08 [alert] 5836#0: *4 connect() failed (88: Socket operation on non-socket) while connecting to upstream, client: 10.xxx.154.yyy,
   server: localhost, request: "GET /zmq HTTP/1.1", upstream: "http://tcp://127.0.0.1:5555/zmq", host: "dans-ubuntu-8375.lvs01.dev.xxx.com"
```

Any ideas why the two protocols on the upstream value? I tried changing the proxy_passs to tcp:// or leaving it off, and got config loading errors. Any remedies appreciated.
Thanks, Dan

**Comment by PiotrSikora** [comment]
Unfortunately, due to the API changes in both ZeroMQ and nginx, this module doesn't really work right now... The good news is that I have updated version in my local tree (along with some other changes) that makes it work again, I just didn't have time to clean it up and commit just yet.

**Comment by beckerdo** [comment]
Hello, has progress been made toward this push? I am interested in seeing our Nginx instance talking to our HTTP handlers. Thanks for all your efforts.

**Comment by neloe** [comment]
I too am interested in this fix.  What is the status; or, more importantly, what changes in the nginx/zmq apis caused this to happen?
Thank you!

**Comment by chaseholland** [comment]
I'm also interested in this fix. Any updates?

**Comment by mikeaorlando** [comment]
This would be so useful if it worked.

**Comment by PiotrSikora** [comment]
Fixed in 08eafd5a93970a805d479312c647e0c658b7ace0.

Sorry for the delay, I've been rather busy with other stuff lately.
